### DISCUSSION TITLE
CORS issue with Discussions API

### DISCUSSION THREAD

**Discussion by laymonage** [community/community#3622] [Discussions] [answered]
It is currently impossible to call the GraphQL API for Discussions from the web browser.

The API requires you to set the `GraphQL-Features: discussions_api` header. However, during the preflight request, the server responds with the following headers:

```
...
Access-Control-Allow-Headers: Authorization, Content-Type, If-Match, If-Modified-Since, If-None-Match, If-Unmodified-Since, Accept-Encoding, X-GitHub-OTP, X-Requested-With, User-Agent
...
```

As the `GraphQL-Features` header isn't included, the browser won't send the API request.

Will this be fixed?

**Comment by bifacil** [comment]
Yes.

Right now is  impossible to call the GraphQL API for Discussions from the web browser.

Is this the place to report the issue? Who can fix this?

**Comment by bifacil** [comment]
Hi @evi-liu, 

I don't know if you are the right person to address this. Sorry if not.

Is possible to call the GraphQL API for Discussions from the web browser? Is this the intended behaviour?

**Comment by mktcode** [comment]
@evi-liu Any update on this?

**Reply by evi-liu** [reply]
Yes! Please see the update here https://github.com/github/feedback/discussions/3622#discussioncomment-821148

**Comment by smashwilson** [comment, marked as answer]
As of yesterday, the `GraphQL-Features: discussions_api` header is no longer necessary to access Discussions-related parts of the schema, so this should work now.

**Reply by laymonage** [reply]
Thank you very much, I can confirm that it works now!

**Reply by mktcode** [reply]
Awesome! :)

**Reply by bifacil** [reply]
Thanks!

**Comment by Estelle544** [comment]
💘
